src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -406,25 +406,29 @@ DeleteAllReferenceTablePlacementsFromNode(char *workerName, uint32 workerPort)
 	referenceTableList = SortList(referenceTableList, CompareOids);
 	foreach(referenceTableCell, referenceTableList)
 	{
+		GroupShardPlacement *placement = NULL;
+		StringInfo deletePlacementCommand = makeStringInfo();
+
 		uint32 workerGroup = GroupForNode(workerName, workerPort);
 
 		Oid referenceTableId = lfirst_oid(referenceTableCell);
 		List *placements = GroupShardPlacementsForTableOnGroup(referenceTableId,
 															   workerGroup);
-		GroupShardPlacement *placement = (GroupShardPlacement *) linitial(placements);
-
-		uint64 shardId = placement->shardId;
-		uint64 placementId = placement->placementId;
+		if (list_length(placements) == 0)
+		{
+			/* this happens if the node was previously disabled */
```

Comment:
There are probably many ways to induce crashes by manually messing with metadata or catalog tables, but ok.